swfiles/sw_spec2MDHisto.m

```diff
@@ -246,27 +250,32 @@ function writeNXlog(filename,log_pth,log_nm,value,units)
     %proj_out = proj(:);
     hkls = dstruct.hkl;
     hkls_sz = size(hkls);
+    % determine the direction where the hkl changes
     dir_vec = hkls(:,hkls_sz(2))-hkls(:,1);
+    dir_vec = dir_vec/norm(dir_vec);
     %qout = hkls'/dir_vec';
     D={};
+    % loop through each of the projection directions
     for idx=1:3
-        procjv = proj(:,idx)/norm(proj(:,idx));
-          
+        qidx = idx; 
```

Comment:
I'm not sure this is needed since `qidx` is always the same as `idx`...

Reply:
Yes the fix was so it would go from -3 to 1 as expected.  I updated to remove the redundancy in the variable.  I chose to keep qidx and remove idx as it is a more descriptive.